Extension/src/Debugger/lldb-dap.ts

```diff
@@ -0,0 +1,229 @@
+                    note(localize('lldb-dap.enabled', "The {0} debugger is enabled via '{1}'", 'lldb-dap', lldbDapPath));
+                    return lldbDapPath;
+                }
+            }
+        }
+    }
+    log(localize('lldb-dap.notfound', "Unable to find a working '{0}' adapter. See: {1}", 'lldb-dap', TroubleshootingLldbDap));
+    note(localize('lldb-dap.notfound', "Unable to find a working '{0}' adapter. See: {1}", 'lldb-dap', TroubleshootingLldbDap));
+    return lldbDapPath;
+}
+
+/** Calls searchForLldbDap, but only runs one search at a time, subsequent calls return the current search.
+ * @returns The path to the lldb dap executable or undefined if it was not found.
+ */
+export async function findLldbDap() {
+    // If we already have a path, return it.
+    if (lldbDapPath) {
+        return lldbDapPath;
+    }
+
+    try {
+        // Only run one search at a time, so if we are already searching, return that.
+        return searching ??= searchForLldbDap();
+    } finally {
+        // When it's done searching, clear the promise so that if it's not found we can try again.
+        void searching?.then(() => searching = undefined);
+    }
+}
+
+/** Check if the given path is a valid lldb dap executable. (actually runs it to check).
+ * @param lldbDap The path to the lldb dap executable.
+ * @returns true If the path is valid, false otherwise.
+ */
+export async function isValidLldbDap(lldbDap: string | undefined) {
+    if (lldbDap) {
+        if (await isExecutable(lldbDap)) {
+            const proc = await spawnChildProcess(lldbDap, ['--help']);
+            if (proc.succeeded && proc.output.includes('USAGE')) {
+                return true;
+            }
+
+            log(localize('lldb-dap.not.valid', "The lldb-dap binary at '{0}' does not appear to be functional. See: {1}", lldbDap, TroubleshootingLldbDap));
+        }
+    }
+    return false;
+}
+
+/** Translates the cpplldb configuration to the lldb dap configuration.
+ * Note: this modifies the existing configuration object in place.
+ *
+ * @param config The cpplldb configuration to translate.
+ * @returns The translated lldb dap configuration.
+ */
+export function translateToLldbDap(config: CppDebugConfiguration) {
+    // Adapt the cpplldb config to the lldb-dap config.
+    if (config.type !== DebuggerType.cpplldb) {
+        throw new Error(`Invalid config type ${config.type} for lldb-dap`);
+    }
+
+    // Translate environment to env.
+    if (config.environment) {
+        // 'config.environment' is an array of { "name":<string>, "value":<string> } objects.
+        // lldb-dap expects an object of { <name>:<value>, ... }.
+        const env: { [key: string]: string } = {};
+        for (const each of config.environment) {
+            if (typeof each.name === 'string' && typeof each.value === 'string') {
+                env[each.name] = each.value;
+            }
+        }
+        delete config.environment;
+    }
+

```

Comment:
You also have `envFile` enabled in the `package.json`, you will also want to read that file and convert it into `env`.

Reply:
`envFile` is already handled prior to it getting here (it loads it and converts it to the environment block